src/nodes/lighting/ShadowNode.js

```diff
@@ -569,7 +569,11 @@ class ShadowNode extends ShadowBaseNode {
 		const depthVersion = shadowMap.depthTexture.version;
 		this._depthVersionCached = depthVersion;
 
-		shadow.camera.layers.mask = camera.layers.mask;
+		if ( shadow.camera.layers.test( 0 ) === true ) {
```

Comment:
I think `test()` is expecting a Layer as a parameter instead of a number, it would also be necessary to restore the previous value after rendering.

Reply:
Ah yes I forgot layers use so sort of Bit Mask. I guess `( shadow.camera.layers.mask & 0xFFFFFFFE ) !== 0` is the right way to check if any layer si set then!

Regarding `shadow.camera.layers.mask = camera.layers.mask;` was always here and never restored. I guess I will add it too.

Reply:
Should be good now!